drivers/charger/sbs_charger/sbs_charger.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sbs_sbs_charger
+
+#include "sbs_charger.h"
+
+#include <zephyr/drivers/charger.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(sbs_charger);
+
+static int sbs_cmd_reg_read(const struct device *dev, uint8_t reg_addr, uint16_t *val)
+{
+	const struct sbs_charger_config *cfg;
+	uint8_t i2c_data[2];
+	int status;
+
+	cfg = dev->config;
+	status = i2c_burst_read_dt(&cfg->i2c, reg_addr, i2c_data, ARRAY_SIZE(i2c_data));
+	if (status < 0) {
+		LOG_ERR("Unable to read register");
+		return status;
+	}
+
+	*val = sys_get_le16(i2c_data);
+
+	return 0;
+}
+
+static int sbs_charger_get_prop(const struct device *dev, struct charger_get_property *prop)
```

Comment:
This looks good given the API as it stands now, but we probably will change it given the comments on the API header.

Reply:
Understood